src/jmap-email/jmap_email/helpers.py

```
@@ -0,0 +1,184 @@
+
+def msgid_chain(ids: Any) -> str:
+    """Reassemble a JMAP ``String[]`` of msg-ids into the angle-bracketed
+    space-separated wire form (e.g. ``"<a@x> <b@x>"``). Strict-typed:
+    see :func:`first_msgid` — only a list of strings is accepted."""
+    if not isinstance(ids, list) or not ids:
+        return ""
+    out: list[str] = []
+    for v in ids:
+        if not isinstance(v, str) or not v:
+            continue
+        v = v.strip()
```

Ruff (0.15.15)

[warning] `for` loop variable `v` overwritten by assignment target

(PLW2901)